cli/js/compiler_host.ts

```diff
@@ -6,8 +6,20 @@ import { cwd } from "./dir.ts";
 import { assert, notImplemented } from "./util.ts";
 import * as util from "./util.ts";
 
+/** Specifies the target that the host should use to inform the TypeScript
+ * compiler of what types should be used to validate the program against. */
+export enum CompilerHostTarget {
+  /** The main isolate library, where the main program runs. */
+  Main = "main",
+  /** The runtime API library. */
+  Runtime = "runtime",
+  /** The worker isolate library, where worker programs run. */
+  Worker = "worker"
+}
+
```

Comment:
So...
- `Runtime` includes Deno runtime declaration
- `Main` includes Deno runtime declaration as well as Window declaration.
- `Worker` includes Deno runtime and Worker declaration

If this is correct, that's just perfect for problem at hand!

Reply:
At the moment, runtime and main are the same, in that they both include Window. I suspect in the future we may want to vary that. Your statement the rest is valid.